src/umple/actions/generate.ts

```diff
@@ -17,16 +18,51 @@ export async function generate() {
 
     const format = await vscode.window.showQuickPick(GENERATE_LANGS);
 
+    if(!format){
+        vscode.window.showInformationMessage("No language is selected");
+        return;
+    }
     if (!format || !editor.document.uri) {
         vscode.window.showInformationMessage("Error occurred format");
         return;
     }
+    
+    var containsError = Lint._storage.getValue("containsCompileError");
 
-    const res = await umpleAPI.generate(editor.document.uri, format);
-    umpleLint.lintFile(editor.document.uri, res);
-    if (res[0].state === 'success' && res[0].message) {
-        vscode.window.showInformationMessage(res[0].message || '');
-    }
+    if(containsError){
+        vscode.window
+        .showInformationMessage(
+          "Your umple code has an error. Do you want to generate the code based on the last successful compile?",
+          ...["Yes", "No"]
+        )
+        .then(async (resp) => {
+          if (resp === "No") {
+            return;
+          }else{
+            const res = await umpleAPI.generate(editor.document.uri, format);
+            umpleLint.lintFile(editor.document.uri, res);
+            if (res[0].state === 'success' && res[0].message) {
+                var extension =  format == 'Ruby'? 'rb' :format=='Umple'?'ump':format.toLowerCase();
+                var message = res[0].message.replace('.ump','.'+ extension);
+                vscode.window.showInformationMessage(message || '');
+            }
+            if(res[1].state == 'error' && res[1].message){
+                vscode.window.showInformationMessage( res[1].message || '');
+            }
+          }
+        });
+    }else{
+        const res = await umpleAPI.generate(editor.document.uri, format);
+        umpleLint.lintFile(editor.document.uri, res);
+        if (res[0].state === 'success' && res[0].message) {
+            var extension =  format == 'Ruby'? 'rb' :format=='Umple'?'ump':format.toLowerCase();
+            var message = res[0].message.replace('.ump','.'+ extension);
+            vscode.window.showInformationMessage(message || '');
+        }
+        if(res[1].state == 'error' && res[1].message){
```

Comment:
believe I had a comment about this in the pr you created previously.

```
if(containsError) {
  const resp = await showInfoMessage();
  if(resp ==="NO") {
      return;
   }
}

// generate file
```

also since generate is already an async function, would prefer if we don't use then

Reply:
Changed and pushed. Would you please review that latest code?